### PR TITLE
Bookmark multi-select

### PR DESCRIPTION
**Pull request by bsclifton**
- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).

## Updates required to sortableTable in order to support multi-drag
- sortableTable selection is now state based (no longer queries DOM)
- code was refactored (logic should be exactly the same) to support being state based
- if multiple items are selected, a different drag handler is applied (so all rows can be sent, not just one)
- if multiple items are selected and there is a drag handler, a dragImage (stack of papers) will be applied on cursor

## Updated functionality on about:bookmarks:
- Existing about:history functionality (including multiple table selection) works great
- about:bookmarks now handle multi-select (for context menus). You can delete multiple entries, etc
- You can select and move multiple items within the same folder OR into a new folder
- The folder view (left) will highlight itself orange when a bookmark (or bookmarks) or folder is dragged over it. This also bolds the text and changes icon to an open folder.

Fixes https://github.com/brave/browser-laptop/issues/1005

## Includes follow up of feedback for
https://github.com/brave/browser-laptop/pull/4834#pullrequestreview-4472932
https://github.com/brave/browser-laptop/pull/4488#pullrequestreview-2648112

## Auditors
@darkdh, @bbondy 

## Screenshots
cc @bradleyrichter 

**multiple selection**
multi-select

**multiple items dragged (notice cursor)**
multiple-drag

**context menu for multiple selected items (ex: delete all selected)**
context-menu

### CODE REVIEW

**Comment by darkdh** [comment]
lgtm

File: js/components/sortableTable.js [hunk 1]
```diff
@@ -17,95 +17,197 @@ tableSort.extend('number', (item) => {
...
+      selection: Immutable.Set()
+    }
   }
   componentDidMount (event) {
     return tableSort(this.table)
   }
-  onClick (e) {
+  /**
+   * If you want multi-select to span multiple tables, you can
+   * provide a single React.Component as the state owner.
```
Comment (bsclifton) [inline js/components/sortableTable.js]: done- thanks :smile:

File: js/about/history.js [hunk 2]
```diff
@@ -31,7 +34,7 @@ class DeleteHistoryEntryButton extends ImmutableComponent {
     if (e && e.preventDefault) {
       e.preventDefault()
     }
-    // BSCTODO: ...
+    // BSCTODO: delete the selected entry
```
Comment (bbondy) [inline js/about/history.js]: todo format is TODO(bsclifton): delete the selected entry

**Comment by bbondy** [comment]
++ :D

**Pull request merged**